File: NbaTracker.Api/NbaTracker.Common/DataContracts/Game.cs
using NbaTracker.Common.Enums;

namespace NbaTracker.Common.DataContracts;

public class Game
{
  public required string GameId { get; set; }
  public GameStatus GameStatus { get; set; }
  public required string GameStatusText { get; set; }
  public int Period { get; set; }
  public required DateTime GameTime { get; set; }
  
  public required Team HomeTeam { get; set; }
  public required Team AwayTeam { get; set; }
  
  public required GameLeader HomeGameLeader { get; set; }
  public required GameLeader AwayGameLeader { get; set; }
}

File: nba-tracker-app/src/app/DataContracts/Game.ts
import { GameStatusEnum } from './enums/GameStatusEnum';
import { Team } from './Team';
import { GameLeader } from './GameLeader';

export interface Game {
  gameId: string;
  gameStatus: GameStatusEnum;
  gameStatusText: string;
  period: number;
  gameTime: Date;

  homeTeam: Team;
  awayTeam: Team;

  homeGameLeader: GameLeader;
  awayGameLeader: GameLeader;
}
